client/http-client/src/client.rs

```diff
@@ -77,10 +82,20 @@ impl HttpClientBuilder {
 		self
 	}
 
+	/// Adds a custom header, which is sent with every request
+	pub fn add_header(mut self, key: &str, value: &str) -> Result<Self, Error> {
```

Comment:
it's because two different `http libraries` are used, soketto is using to `httparse::Header` which isn't as pedantic as hyper::HeaderMap is. 

that's why I was leaning into exposing `fn add_headers(header: HeaderMap) -> Self` for both instead which makes things more uniform and quite obvious how it works.

Reply:
Ah I see! I guess the issue there is that the ws-client doesn't (afaict) depend on hyper, so it would be a shame to add it as a dependency just for that type?